engine/src/main/java/com/datatorrent/stram/ResourceRequestHandler.java

```diff
@@ -108,6 +113,23 @@ public void updateNodeReports(List<NodeReport> nodeReports)
     }
   }
 
+  public List<String> getNodesExceptHost(List<String> hostNames)
```

Comment:
Does hostNames always contain one element? The contains operation is expensive...